modules/shared/src/main/scala/io/constellationnetwork/schema/delegatedStake.scala

```diff
@@ -105,6 +107,34 @@ object delegatedStake {
     pendingWithdrawals: List[DelegatedStakeInfo]
   )
 
+  @derive(decoder, encoder, eqv, show)
+  case class DelegatedStakeAccount(
+    activeStakes: List[(Signed[UpdateDelegatedStake.Create], SnapshotOrdinal)],
+    accruedRewards: Map[Address, Balance]
+  )
+
+  object DelegatedStakeAccount {
+    val empty: DelegatedStakeAccount = DelegatedStakeAccount(List.empty, Map.empty[Address, Balance])
+
+    implicit val delegatedStakeWithBalanceOrdering: Ordering[DelegatedStakeAccount] =
+      new Ordering[DelegatedStakeAccount] {
+        def compare(x: DelegatedStakeAccount, y: DelegatedStakeAccount): Int = {
+          val xTotalBalance = x.accruedRewards.values.foldLeft(Balance.empty.value.value)((l, r) => l + r.value.value)
+          val yTotalBalance = y.accruedRewards.values.foldLeft(Balance.empty.value.value)((l, r) => l + r.value.value)
+
+          Ordering[Long].compare(xTotalBalance, yTotalBalance)
```

Comment:
Why the discrepancy in ordering in state? It seems like createdAt ordering for PendingWithdrawal makes a lot more sense to me for an internal state ordering. Is this ordering applied on the state object? Or for the REST API?

Reply:
The original implementation used a `SortedMap`, this was simply a straightforward method to provide an `Ordering` instance as needed for that data structure.

Reply:
Okay, I think that works. I'm pretty sure the sorting on state is never used anywhere. And since we're reconstructing the state map everywhere it's not really an issue. I could see this potentially being an issue if we eventually migrate the state to a separate store for the proof construction, because it shuffles based not on acceptance order, but that really doesn't matter yet so this is fine I think, can resolve this.